truffleruby/src/main/java/org/truffleruby/language/dispatch/CachedDispatchNode.java

```diff
@@ -60,10 +63,17 @@ public CachedDispatchNode(
         this.next = next;
     }
 
-    public void replaceSendingChild() {
+    public void replaceSendingFrame() {
         CompilerAsserts.neverPartOfCompilation("Dispatch nodes should not be altered after compilation.");
-        if (!needsCallerFrame) {
-            needsCallerFrame = true;
+        if (!sendsFrame) {
+            sendsFrame = true;
+        }
+    }
+
+    public void replaceSendingCallerFrame(CallerFrameAccess access) {
+        CompilerAsserts.neverPartOfCompilation("Dispatch nodes should not be altered after compilation.");
```

Comment:
I missed this before. What exactly does it mean? What happens if methods are monkey patched?